base/random/generation.jl

```diff
@@ -0,0 +1,321 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+# Uniform random generation
+
+## from types: rand(::Type, [dims...])
+
+### GLOBAL_RNG fallback for all types
+
+@inline rand(T::Type) = rand(GLOBAL_RNG, T)
+
+### random floating point numbers
+
+@inline rand(r::AbstractRNG=GLOBAL_RNG) = rand(r, CloseOpen)
+
+@inline rand_generic(r::AbstractRNG, ::Type{Float64}) = rand(r, CloseOpen)
+
+rand_generic(r::AbstractRNG, ::Type{Float16}) =
+    Float16(reinterpret(
+        Float32, (rand_ui10_raw(r) % UInt32 << 13) & 0x007fe000 | 0x3f800000) - 1)
+
+rand_generic(r::AbstractRNG, ::Type{Float32}) =
+    reinterpret(Float32, rand_ui23_raw(r) % UInt32 & 0x007fffff | 0x3f800000) - 1
+
+### random integers
+
+rand_ui10_raw(r::AbstractRNG) = rand(r, UInt16)
+rand_ui23_raw(r::AbstractRNG) = rand(r, UInt32)
+@inline rand_ui52_raw(r::AbstractRNG) = reinterpret(UInt64, rand(r, Close1Open2))
+@inline rand_ui52(r::AbstractRNG) = rand_ui52_raw(r) & 0x000fffffffffffff
+
+### random complex numbers
+
+rand(r::AbstractRNG, ::Type{Complex{T}}) where {T<:Real} = complex(rand(r, T), rand(r, T))
+
+### random Char values
+
+# returns a random valid Unicode scalar value (i.e. 0 - 0xd7ff, 0xe000 - # 0x10ffff)
+function rand(r::AbstractRNG, ::Type{Char})
+    c = rand(r, 0x00000000:0x0010f7ff)
+    (c < 0xd800) ? Char(c) : Char(c+0x800)
+end
+
+### arrays of random numbers
+
+function rand!(r::AbstractRNG, A::AbstractArray{T}, ::Type{X}=T) where {T,X}
+    for i in eachindex(A)
+        @inbounds A[i] = rand(r, X)
+    end
+    A
+end
+
+rand!(A::AbstractArray{T}, ::Type{X}=T) where {T,X} = rand!(GLOBAL_RNG, A, X)
+
+rand(r::AbstractRNG, dims::Dims)       = rand(r, Float64, dims)
+rand(                dims::Dims)       = rand(GLOBAL_RNG, dims)
+rand(r::AbstractRNG, dims::Integer...) = rand(r, Dims(dims))
+rand(                dims::Integer...) = rand(Dims(dims))
+
+rand(r::AbstractRNG, T::Type, dims::Dims)                    = rand!(r, Array{T}(dims))
+rand(                T::Type, dims::Dims)                    = rand(GLOBAL_RNG, T, dims)
+rand(r::AbstractRNG, T::Type, d1::Integer, dims::Integer...) = rand(r, T, Dims((d1, dims...)))
+rand(                T::Type, d1::Integer, dims::Integer...) = rand(T, Dims((d1, dims...)))
```

Comment:
hopefully equivalent, but any reason in particular for changing this from the previous `rand(T::Type, d1::Integer, dims::Integer...) = rand(T, tuple(Int(d1), convert(Dims, dims)...))` ?